ci/ansible/playbooks/os/redhat/upgrade_release.yml

```diff
@@ -0,0 +1,378 @@
+              shell: "subscription-manager list --installed|awk '/Status:/ { print $2 }'"
+              register: subscription_status
+
+            - name: Check that upgraded version remains correctly subscribed
+              assert:
+                that:
+                  - subscription_version.stdout == "8.4"
+                  - subscription_status.stdout == "Subscribed"
+                quiet: true
+
+    - name: Remove packages from the dnf exclude list # Populated by leapp during upgrade
+      ini_file:
+        path: /etc/dnf/dnf.conf
+        section: main
+        option: exclude
+        value: ''
+
+    - name: Remove leapp_resume service unit
+      block:
+        - name: Disable leapp_resume service # This unit may remain even when the leapp packages are removed
+          systemd:
+            name: leapp_resume
+            state: stopped
+            enabled: false
+            daemon-reload: true
+            force: true
+
+        - name: Remove leapp_resume unit file # Package removal does not remove this units automatically
+          file:
+            path: /etc/systemd/system/leapp_resume.service
+            state: absent
+
+    - name: Remove leapp packages
+      package:
+        name: "leapp*" # Leapp packages may have different nomenclature in RHEL8
+        state: absent
+
+    - name: Remove leapp configuration and log files
+      file:
+        path: /etc/leapp
+        state: absent
+
+    - name: Remove remaining RHEL 7 packages
+      block:
+        - name: Register remaining RHEL7 packages
+          shell: "rpm -qa | grep -e '\\.el7' | grep -vE '^(gpg-pubkey|libmodulemd|katello-ca-consumer)'"
+          register: remaining_rhel7_packages
+          ignore_errors: true
+
+        - name: Remove remaining RHEL7 packages
+          when: not remaining_rhel7_packages.failed
+          dnf:
+            name: "{{ remaining_rhel7_packages.stdout_lines }}"
+            state: absent
+
+        - name: Determine RHEL7 old kernel versions
+          command: "ls -1d *.el7*"
+          args:
+            chdir: /lib/modules
+          register: old_kernel_versions
+          ignore_errors: true
+
+        - name: Remove weak modules from old kernels
+          when: not old_kernel_versions.failed
+          shell: "[ -x /usr/sbin/weak-modules ] && /usr/sbin/weak-modules --remove-kernel {{ item }}"
+          loop: "{{ old_kernel_versions.stdout_lines }}"
+
+        - name: Remove old kernels from bootloader entries
+          when: not old_kernel_versions.failed
+          command: "/bin/kernel-install remove {{ item }} /lib/modules/{{ item }}/vmlinuz"
+          loop: "{{ old_kernel_versions.stdout_lines }}"
+
+    - name: Restore root login only if it was originally enabled
+      when: sshd_config.changed
+      lineinfile:
+        dest: /etc/ssh/sshd_config
+        regexp: '^PermitRootLogin no'
+        line: 'PermitRootLogin yes'
+        state: present
+
+    - *RESTART_SSHD
```

Comment:
What is this?

Reply:
*RESTART_SSHD executes &RESTART_SSHD defined in line 140